listenbrainz/mbid_mapping/mapping/dump.py

```diff
@@ -0,0 +1,219 @@
+""" This module contains data dump creation and import functions.
+
+Read more about the data dumps in our documentation here:
+https://listenbrainz.readthedocs.io/en/production/dev/listenbrainz-dumps.html
+"""
+
+# listenbrainz-server - Server for the ListenBrainz project
+#
+# Copyright (C) 2017 MetaBrainz Foundation Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+
+import os
+import shutil
+import subprocess
+import tarfile
+import tempfile
+from datetime import datetime
+from typing import Optional
+
+import sqlalchemy
+from flask import current_app, render_template
+from psycopg2.sql import SQL
+
+from listenbrainz import DUMP_LICENSE_FILE_PATH
+from brainzutils import musicbrainz_db
+from listenbrainz.db.dump import _escape_table_columns
+from listenbrainz.utils import create_path
+
+
+PUBLIC_TABLES_MAPPING = {
+    'mapping.canonical_musicbrainz_data': {
+        'filename': 'canonical_musicbrainz_data.csv',
+        'columns': (
+            'id',
+            'artist_credit_id',
+            'artist_mbids',
+            'artist_credit_name',
+            'release_mbid',
+            'release_name',
+            'recording_mbid',
+            'recording_name',
+            'combined_lookup',
+            'score',
+            'year',
+        ),
+    },
+    'mapping.canonical_recording_redirect': {
+        'filename': 'canonical_recording_redirect.csv',
+        'columns': (
+            'recording_mbid',
+            'canonical_recording_mbid',
+            'canonical_release_mbid'
+        )
+    }
+}
+
+
+def dump_postgres_db(location, dump_time=datetime.today()):
+    """ Create a MusicBrainz canonical metadata postgres database dump in the specified location
+
+        Arguments:
+            location: Directory where the final dump will be stored
+            dump_time: datetime object representing when the dump was started
+
+        Returns:
+            the path to the dumped data
+    """
+    current_app.logger.info('Beginning dump of PostgreSQL database...')
+    current_app.logger.info('dump path: %s', location)
+
+    current_app.logger.info('Creating dump of canonical metadata data...')
+    try:
+        dump_location = create_mapping_dump(location, dump_time)
+    except IOError as e:
```

Comment:
Nit: I know we do this in normal dumps as well but I think multiple excepts are not useful because we do the exactly same thing and `exc_info` would record the exception info anyway.